pkg/testingutil/k8sClient.go

```diff
@@ -0,0 +1,32 @@
+package testingutil
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type FakeK8sClient struct {
+	client.Client         // To satisfy interface; override all used methods
```

Comment:
Thinking how would for example odo use this client? I checked their kclient, it looks like they have a different client? Would they need to implement the Get() as well?

Reply:
yes they have a different client, since they self-defined the client struct. this client we are using here is from the standard k8 controller-runtime pkg `"sigs.k8s.io/controller-runtime/pkg/client"`

to get the k8sclient, can just simply run this with the kube config
```
k8sclient, err := client.New(config.GetConfig(), client.Options{})
```

They don't need to implement the Get(), as it is implemented by the standard pkg itself. We override this get method just for test, because we don't want to create a real kclient for unit test.